test/e2e/apps/job.go

```diff
@@ -465,6 +465,78 @@ done`}
 		gomega.Expect(job.Status.Succeeded).Should(gomega.Equal(int32(2)))
 	})
 
+	/*
+		Testcase: Terminate job execution when the maxFailedIndexes is exceeded
+		Description: Create an indexed job with backoffLimitPerIndex and maxFailedIndexes.
+		Verify the job execution is terminated as soon as the number of failed
+		indexes exceeds maxFailedIndexes.
+	*/
+	ginkgo.It("should terminate job execution when the number of failed indexes exceeds maxFailedIndexes", func(ctx context.Context) {
+		// we use parallelism=1 to make sure in the asserts only one pod was created
+		parallelism = int32(1)
+		ginkgo.By("Creating an indexed job with backoffLimit per index and maxFailedIndexes")
+		job := e2ejob.NewTestJob("fail", "with-max-failed-indexes", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
```

Comment:
side note: having these global variables for `parallelism`, `completions` and `backoffLimit` make the tests very hard to read. I need to jump from one place to another to check what the values are. We should get rid of them in a follow up.

Reply:
sgtm

Reply:
Actually, I just realized this `parallelism = int32(1)` was a bug that could override the parallelism for the next test to run. This caused this failure which I also reproduced locally: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121633/pull-kubernetes-e2e-gce/1719394173829779456. Fixed now by  `parallelism := int32(1)`. However, its better to avoid the global vars. I will open an issue for this.

Reply:
Opened the cleanup issue for now, maybe some folks will want to grab: https://github.com/kubernetes/kubernetes/issues/121689